README.md

```diff
@@ -21,63 +21,110 @@ apply plugin: "com.vanniktech.maven.publish"
 
 Information: [This plugin is also available on Gradle plugins](https://plugins.gradle.org/plugin/com.vanniktech.maven.publish)
 
-### Snapshots
+Snapshots can be found [here](https://oss.sonatype.org/#nexus-search;quick~gradle-maven-publish-plugin).
 
-Can be found [here](https://oss.sonatype.org/#nexus-search;quick~gradle-maven-publish-plugin). Current one is:
+### Setting properties
 
-```groovy
-classpath 'com.vanniktech:gradle-maven-publish-plugin:0.9.0-SNAPSHOT'
+To configure the coordinates of your published artifacts as well as the POM this plugin 
+uses Gradle properties. It's generally recommended to set them in your `gradle.properties`
+file.
+
+There are 3 required properties:
+```
+GROUP=com.test.mylibrary
+POM_ARTIFACT_ID=mylibrary-runtime
+VERSION_NAME=3.0.5
+```
+
+In addition to that there are some optional properties to give more details about your project:
+
+```
+POM_NAME=My Library
+POM_DESCRIPTION=A description of what my library does.
+POM_INCEPTION_YEAR=2020
+
+POM_URL=https://github.com/username/mylibrary/
+POM_SCM_URL=https://github.com/username/mylibrary/
+POM_SCM_CONNECTION=scm:git:git://github.com/username/mylibrary.git
+POM_SCM_DEV_CONNECTION=scm:git:ssh://git@github.com/username/mylibrary.git
+
+POM_LICENCE_NAME=The Apache Software License, Version 2.0
+POM_LICENCE_URL=https://www.apache.org/licenses/LICENSE-2.0.txt
+POM_LICENCE_DIST=repo
+
+POM_DEVELOPER_ID=username
+POM_DEVELOPER_NAME=User Name
+POM_DEVELOPER_URL=https://github.com/username/
 ```
 
-## Configuration
+This Gradle plugin is using itself to publish any of the updates and sets the Gradle properties in 
+this [gradle.properties](gradle.properties).
+
+In multi module projects you can set most properties in the root `gradle.properties` file and
+then only set the module specific ones in the submodules. For example if you have two modules 
+called `runtime` and `driver` you could only set `POM_ARTIFACT_ID` and `POM_NAME` in 
+`<project-dir>/runtime/gradle.properties` and `<project-dir>/driver/gradle.properties` while sharing 
+the rest by putting them into `<project-dir>/gradle.properties`.
+
+### Where to upload to
+
+Without any further configuration the plugin has two tasks. `uploadArchives` which will upload
+to Sonatype OSS (Maven Central) by default. To publish to the local maven repository on your 
+machine (`~/m2/repository`) there is `installArchives`. 
 
-***Uploading:***
+The username and password for Sonatype OSS can be provided as Gradle properties or environment
```

Comment:
Let's mention which properties need to be set here for Gradle

Reply:
Do you mean something other than `SONATYPE_NEXUS_USERNAME` and `SONATYPE_NEXUS_PASSWORD` which are mentioned in the next line?

Reply:
Oh right they're both sharing the same name. Then never mind